cmd/kubeadm/app/discovery/https/https.go

```diff
@@ -29,7 +30,8 @@ import (
 // securely to the API Server using the provided CA cert and
 // optionally refreshes the cluster-info information from the cluster-info ConfigMap
 func RetrieveValidatedClusterInfo(httpsURL string) (*clientcmdapi.Cluster, error) {
-	response, err := http.Get(httpsURL)
+	client := &http.Client{Transport: netutil.SetOldTransportDefaults(&http.Transport{})}
```

Comment:
why not `SetTransportDefaults`?

Reply:
SetTransportDefaults does http2 initialization, which in most cases here is not expected.
Even SetOldTransportDefaults does a bit more than expected, but setting defaults for Dialer and TLSHandshakeTimeout is safe enough to not be worried about.